tests/model_serving/model_server/components/kserve_dsc_deployment_mode/utils.py

```diff
@@ -67,3 +68,86 @@ def patch_dsc_default_deployment_mode(
             deployment_mode=request_default_deployment_mode,
         )
         yield dsc_resource
+
+
+def wait_for_service_config_in_cm(config_map: ConfigMap, service_config: str) -> None:
+    """
+    Wait for service config to be set in inferenceservice-config configmap.
+
+    Args:
+        config_map (ConfigMap): ConfigMap object
+        service_config (str): Service configuration type ("Headless" or "Headed")
+
+    Raises:
+        TimeoutExpiredError: If service config value is not set in configmap
+    """
+    LOGGER.info(f"Waiting for {service_config} service config to be set in {config_map.name} configmap")
+    expected_value = service_config == "Headless"
+    for sample in TimeoutSampler(
+        wait_timeout=Timeout.TIMEOUT_5MIN,
+        sleep=5,
+        func=lambda: config_map.instance.data,
+    ):
+        if sample:
+            try:
+                cm_service_config = json.loads(sample.service)["serviceClusterIPNone"]
```

Comment:
better user `get` here so if the key is missing for some reason, the failure will be on wrong value and not missing key